python/pyspark/ml/regression.py

```diff
@@ -743,6 +743,18 @@ def treeWeights(self):
         """Return the weights for each tree"""
         return list(self._call_java("javaTreeWeights"))
 
```

Comment:
Can we not simply override the method for the case where it is a
classification model? I think that is only for RFClasaifier since GBT
actually returns DTRegreasionModels
On Thu, 5 May 2016 at 18:44, Holden Karau <notifications@github.com> wrote:

> In python/pyspark/ml/regression.py
> <https://github.com/apache/spark/pull/12919#discussion_r62215111>:
>
> > @@ -743,6 +743,18 @@ def treeWeights(self):
> >          """Return the weights for each tree"""
> >          return list(self._call_java("javaTreeWeights"))
> >
>
> So we could do that, the thing is it might be a
> DecisionTreeRegressionModel or a DecisionTreeClassificationModel so we
> probably just want to return a list of DecisionTreeModel instead.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12919/files/6400e9e3d7e096b05475cca548e31aabca8cbaff#r62215111>
>